Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,15 +1,82 @@
-﻿namespace BootCamp.Chapter1
+﻿using System;
+namespace BootCamp.Chapter1
 {
     public static class ArrayOperations
     {
+        // *************private helpers**********
         /// <summary>
+        /// Test that Array is neither null or empty.
+        /// </summary>
+        /// <returns>True if array can be manipulated by standard functions.</returns>
+        /// 
+        private static bool ValidateArrayForManipulation(int[] array)
+        {
+            return ((array != null) && (array.Length != 0));
+        }
+
+        /// <summary>
+        /// Makes the array one index item shorter.
+        /// </summary>
+        /// <returns>A new array with (original length - 1)</returns>
+        /// 
+        private static int[] MakeAShorterArray(int[] array)
+        {
+            return new int[array.Length - 1];
+        }
```

Comment:
This is a bit too less code (or better: logic) to make a new function for.
You did understand what code duplication means, though, which is a great thing. But always think about how useful will it be? However such a function COULD exist, but then probably more in that way:

```csharp
private static int[] ShortenBy(int[] array, int amount)
{
    if (amount > array.Length || amount <= 0)
    {
        return array;
    }
    return new int[array.Length - amount];
}
```